[Term Entry] Java Calendar add() (#3247)
Codecademy/docs: content/java/concepts/calendar/terms/add/add.md

---
Title: '.add()'
Description: 'Adds or subtracts a specified amount of time to/from the given Calendar field.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Calendar'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

The **`.add()`** method of the `Calendar` class is used to manipulate and modify a `Calendar` instance by adding or subtracting a specified amount of time to/from a specific calendar field.

## Syntax

```pseudo
myCalendar.add(field, amount)
```

> - `field`: An integer constant representing the field to which the amount of time should be added or subtracted (e.g.,`Calendar.YEAR`, `Calendar.MONTH`, `Calendar.DAY_OF_MONTH`).
> 
> - `amount`: An integer value representing the amount of time to add or subtract from the specified field. A positive value adds time, and a negative value subtracts time.

Reviewer: ```suggestion
- `field`: An integer constant representing the field to which the amount of time should be added or subtracted (e.g.,`Calendar.YEAR`, `Calendar.MONTH`, `Calendar.DAY_OF_MONTH`).
- `amount`: An integer value representing the amount of time to add or subtract from the specified field. A positive value adds time, and a negative value subtracts time.
```
Remove blank line from list,

## Example

> In the example below, we demonstrate how to use the `.add()` method to manipulate a `Calendar` instance:

Reviewer: ```suggestion
The example below demonstrates how to use the `.add()` method to manipulate a `Calendar` instance:
```

Author: I've made the requested edits. Thanks @SSwiniarski!

---
Title: '.add()'
Description: 'Adds or subtracts a specified amount of time to/from the given Calendar field.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Calendar'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The **`.add()`** method of the `Calendar` class is used to manipulate and modify a `Calendar` instance by adding or subtracting a specified amount of time to/from a specfic Calendar field.

Reviewer 2: ```suggestion
The **`.add()`** method of the `Calendar` class is used to manipulate and modify a `Calendar` instance by adding or subtracting a specified amount of time to/from a specific calendar field.
```

## Syntax

```pseudo
myCalendar.add(field, amount)
```

- `field`: An integer constant representing the field to which the amount of time should be added or subtracted. This can be one of the following constants:

>     - `Calendar.YEAR`: Adds or subtracts the specified number of years.

Reviewer 2: I think these fields are pretty self-explanatory. I would suggest only listing a few as examples of what a field value could be, or replace the existing descriptions with more detailed information that may not be as obvious such as the range for the given field.

    - `Calendar.MONTH`: Adds or subtracts the specified number of months.
    - `Calendar.DAY_OF_MONTH`: Adds or subtracts the specified number of days.
    - `Calendar.HOUR_OF_DAY`: Adds or subtracts the specified number of hours.
    - `Calendar.MINUTE`: Adds or subtracts the specified number of minutes.
    - `Calendar.SECOND`: Adds or subtracts the specified number of seconds.
    - `Calendar.MILLISECOND`: Adds or subtracts the specified number of milliseconds.

- `amount`: An integer value representing the amount of time to add or subtract from the specified field. A positive value adds time, and a negative value subtracts time.

## Example

In the example below, we demonstrate how to use the `.add()` method to manipulate a `Calendar` instance:

```java
import java.util.Calendar;

> public class Main {

Reviewer 2: The result of the operations might be clearer if you include a print call after each change

  public static void main(String[] args) {

    // Create a Calendar instance representing the current date and time
    Calendar myCalendar = Calendar.getInstance();

    // Add 2 years to the current date
    myCalendar.add(Calendar.YEAR, 2);

    // Add 3 months to the current date
    myCalendar.add(Calendar.MONTH, 3);

    // Subtract 10 days from the current date
    myCalendar.add(Calendar.DAY_OF_MONTH, -10);

    // Display the modified date
    System.out.println("Modified Date: " + myCalendar.getTime());
  }
}
```

Output for the above code will vary depending on the current date but might look like:

> ```java

Reviewer 2: ```suggestion
```shell
```

Modified Date: Thu Dec 05 14:23:45 UTC 2025
```

> In this example, we first create a `Calendar` instance representing the current date and time. Then, we use the `.add()` method to manipulate this instance by adding 2 years, 3 months, and subtracting 10 days, resulting in a modified date.

Reviewer 2: ```suggestion
```
